src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/BindSymbolEvaluator.cs

```diff
@@ -0,0 +1,171 @@
+
+            foreach (var task in tasksToRun.Where(t => t.Task.IsFaulted || t.Task.IsCompleted && t.Task.Result == null))
+            {
+                _logger.LogWarning(LocalizableStrings.BindSymbolEvaluator_Warning_EvaluationError, task.Symbol.Name);
+            }
+
+            var successfulTasks = tasksToRun.Where(t => t.Task.IsCompleted && t.Task.Result != null);
+            foreach (var task in successfulTasks)
+            {
+                variableCollection[task.Symbol.Name] = RunnableProjectGenerator.InferTypeAndConvertLiteral(task.Task.Result!);
+            }
+        }
+
+        private async Task<string?> GetBoundValueAsync(string configuredBinding, CancellationToken cancellationToken)
+        {
+            if (string.IsNullOrWhiteSpace(configuredBinding))
+            {
+                throw new ArgumentException($"'{nameof(configuredBinding)}' cannot be null or whitespace.", nameof(configuredBinding));
+            }
+
+            string? prefix = null;
+            string binding = configuredBinding;
+            if (configuredBinding.Contains(":"))
+            {
+                prefix = configuredBinding.Substring(0, configuredBinding.IndexOf(":")).Trim();
+                binding = configuredBinding.Substring(configuredBinding.IndexOf(":") + 1).Trim();
+            }
+            if (string.IsNullOrWhiteSpace(binding))
+            {
+                binding = configuredBinding;
+            }
+
+            IEnumerable<IBindSymbolSource>? sourcesToSearch = null;
+            if (string.IsNullOrWhiteSpace(prefix))
+            {
+                sourcesToSearch = _bindSymbolSources;
+            }
+            else
+            {
+                sourcesToSearch = _bindSymbolSources.Where(s => s.SourcePrefix?.Equals(prefix, StringComparison.OrdinalIgnoreCase) ?? false);
+                if (!sourcesToSearch.Any())
+                {
+                    //if there is no matching sources, then use all the sources with full binding
+                    sourcesToSearch = _bindSymbolSources;
+                    binding = configuredBinding;
+                }
+            }
+
+            var successfulTasks = await RunEvaluationTasks(sourcesToSearch, binding, cancellationToken).ConfigureAwait(false);
+            cancellationToken.ThrowIfCancellationRequested();
+            if (!successfulTasks.Any())
+            {
+                //if nothing is found, try all sources with unparsed values from configuration
+                successfulTasks = await RunEvaluationTasks(_bindSymbolSources, configuredBinding, cancellationToken).ConfigureAwait(false);
+            }
+            cancellationToken.ThrowIfCancellationRequested();
+
+            if (!successfulTasks.Any())
+            {
+                return null;
+            }
+            else if (successfulTasks.Count() == 1)
+            {
+                return successfulTasks.Single().Value;
+            }
+            else
```

Comment:
I still feel that presence of multiple matching sources (despite distinct priorities) is highly suspicious and deserve warning. What do you think?

Reply:
I added more debug logging, I think it's a valid case when there are multiple sources providing the value and then priority is decision maker, so warning (user facing) feels too much details.